tests/csapi/apidoc_room_state_test.go

```diff
@@ -104,26 +107,31 @@ func TestRoomState(t *testing.T) {
 				"preset":     "public_chat",
 			})
 
-			res := authedClient.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "publicRooms"})
-			foundRoom := false
-
-			must.MatchResponse(t, res, match.HTTPResponse{
-				JSON: []match.JSON{
-					match.JSONKeyPresent("chunk"),
-					match.JSONArrayEach("chunk", func(val gjson.Result) error {
-						gotRoomID := val.Get("room_id").Str
-						if gotRoomID == roomID {
-							foundRoom = true
-							return nil
-						}
-						return nil
-					}),
-				},
-			})
-
-			if !foundRoom {
-				t.Errorf("failed to find room with id: %s", roomID)
-			}
+			authedClient.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "publicRooms"},
+				client.WithRetryUntil(time.Second, func(res *http.Response) bool {
+					foundRoom := false
+
+					must.MatchResponse(t, res, match.HTTPResponse{
+						JSON: []match.JSON{
+							match.JSONKeyPresent("chunk"),
+							match.JSONArrayEach("chunk", func(val gjson.Result) error {
+								gotRoomID := val.Get("room_id").Str
+								if gotRoomID == roomID {
+									foundRoom = true
+									return nil
+								}
+								return nil
```

Comment:
This is a slight misuse of matchers, given you always return `nil` it will always succeed even if the room isn't found, and then you side-effect `foundRoom` to determine retry-ness. It's not something that the matchers were designed for, but in practice it seems to read pretty well.

Reply:
You're right, though I will say 'don't blame me, it was there before!' :-P

I suspect this sort of pattern will crop up again, so if you get any better ideas then it may be worth putting them in